### ISSUE TITLE
Single line try catch statement with reasonable limitations.

### ISSUE THREAD

**Issue by MkazemAkhgary** [dotnet/csharplang#1067]
people are arguing that `try catch` with no bracket is ambiguous. blaming such proposals for much less common case that you cant have multiple catch statements that would be ambiguous for compiler.

Just ask your self

 1. How often do you catch different exceptions for single try statement?
 2. How often do you catch single exception for single line try statement?

Be honest. don't rush on this. with 4 years of programming (I know that's not much experience), I would say

 1. not so much, maybe on root of API functions (I did that rarely, usually Exception and log is enough.)
 2. pretty much often. way more often than catching multiple cases.

Every syntax has it drawbacks and benefits. for this syntax I see much more benefit than its drawbacks. Stop arguing ambiguity, if you want multiple catch statements, you don't use this syntax. its not for you.

---

Here is very simple limitation that you have for single line try catch statement.

 - Multiple catch statements are not allowed after bracket-less try statement.

breaking this rule results in compiler error. 

---

In conclusion, each bracket-less try statement, must come with single catch and/or finally statement. the catch/finally statements are not forced to be bracket less. just like how `if ... else {}` is valid.

---

This proposal supports #1015 , #616

**Comment by HaloFour** [comment]
Done:

```cs
try { DoSomething(); } catch(Exception ex) { Handle(ex); }
```

**Comment by MkazemAkhgary** [comment]
@HaloFour you didn't answer my 2 questions. also that is not one line. I know you are playing with me but I don't like it 😞

>try { DoSomething(); } catch(Exception ex) { Handle(ex); }

now I play with you 😈 , what kind of style is this? https://en.wikipedia.org/wiki/Indentation_style

**Comment by HaloFour** [comment]
The questions don't matter.  Theres no real verbosity improvement to dropping the braces.

Indentation styles are guidelines.  C# is not a whitespace sensitive language.  There's zero need to span a simple try over 6 lines.

**Comment by MkazemAkhgary** [comment]
@HaloFour Its zero need but code refactoring tools will do that like very famous tool resharper.

Now that you may say you can change options, but i want allman style for else where

I made two other proposal that are based on this. So removing braces is not the only concern.

**Comment by jnm2** [comment]
> code refactoring tools will do that like very famous tool resharper.

Not if you maintain a .dotSettings file in the repository telling it not to.

**Comment by MkazemAkhgary** [comment]
@jnm2 if I wanted to do that I would prefer
```
try { DoSomething(); } 
catch(Exception ex) { Handle(ex); }
```
for if statements, I prefer brackets when containing statement is long or is very important (you know I mean by _very_ important 😸 ), but I tend to omit them when statement is short.

being able to remove these brackets isn't big deal. that also avoids making me struggle and tune code refactoring options.

```
try DoSomething();
catch(Exception ex) Handle(ex);
```

**Comment by shinsetsu** [comment]
> The questions don't matter. Theres no real verbosity improvement to dropping the braces.
> 
> Indentation styles are guidelines. C# is not a whitespace sensitive language. There's zero need to span a simple try over 6 lines.

I wanna drop braces so theres less characters...    Squirrel....
